pkg/workflow_execution.go

```diff
@@ -889,30 +933,61 @@ func (c *Client) GetWorkflowExecutionMetrics(namespace, uid, podName string) (me
 		return nil, util.NewUserError(codes.NotFound, "Can't get configuration.")
 	}
 
-	s3Client, err = c.GetS3Client(namespace, config.ArtifactRepository.S3)
-	if err != nil {
-		log.WithFields(log.Fields{
-			"Namespace": namespace,
-			"UID":       uid,
-			"PodName":   podName,
-			"Error":     err.Error(),
-		}).Error("Can't connect to S3 storage.")
-		return nil, util.NewUserError(codes.NotFound, "Can't connect to S3 storage.")
-	}
+	switch {
+	case config.ArtifactRepository.S3 != nil:
+		{
+			s3Client, err = c.GetS3Client(namespace, config.ArtifactRepository.S3)
+			if err != nil {
+				log.WithFields(log.Fields{
+					"Namespace": namespace,
+					"UID":       uid,
+					"PodName":   podName,
+					"Error":     err.Error(),
+				}).Error("Can't connect to S3 storage.")
+				return nil, util.NewUserError(codes.NotFound, "Can't connect to S3 storage.")
+			}
 
-	opts := s3.GetObjectOptions{}
+			opts := s3.GetObjectOptions{}
 
-	key := config.ArtifactRepository.S3.FormatKey(namespace, uid, podName) + "/sys-metrics.json"
-	stream, err = s3Client.GetObject(config.ArtifactRepository.S3.Bucket, key, opts)
-	if err != nil {
-		log.WithFields(log.Fields{
-			"Namespace": namespace,
-			"UID":       uid,
-			"PodName":   podName,
-			"Error":     err.Error(),
-		}).Error("Metrics do not exist.")
-		return nil, util.NewUserError(codes.NotFound, "Metrics do not exist.")
+			key := config.ArtifactRepository.S3.FormatKey(namespace, uid, podName) + "/sys-metrics.json"
+			stream, err = s3Client.GetObject(config.ArtifactRepository.S3.Bucket, key, opts)
+			if err != nil {
+				log.WithFields(log.Fields{
+					"Namespace": namespace,
+					"UID":       uid,
+					"PodName":   podName,
+					"Error":     err.Error(),
+				}).Error("Metrics do not exist.")
+				return nil, util.NewUserError(codes.NotFound, "Metrics do not exist.")
+			}
+		}
+	case config.ArtifactRepository.GCS != nil:
+		{
+			ctx := context.Background()
```

Comment:
[These lines](https://github.com/onepanelio/core/pull/400/files#diff-2df54c8de22b7ec30c64350c5cbfd591R966-R987) seem to be repeated a bunch of times, can this be put into one method?

Reply:
Refactored